pkg/sql/descriptor.go

```diff
@@ -226,6 +226,14 @@ func (p *planner) createDatabase(
 func (p *planner) createDescriptor(
 	ctx context.Context, descriptor catalog.MutableDescriptor, jobDesc string,
 ) error {
+	if p.ExecCfg().Codec.ForSystemTenant() && !p.EvalContext().SessionData().Internal &&
+		RestrictAccessToSystemInterface.Get(&p.ExecCfg().Settings.SV) {
```

Comment:
I could see having two methods on the planner `(p *planner ) shouldRestrictSystemInterfaceAccess() bool` that you could use here and in set_cluster_setting.go